test/tests/functional/pbs_power_provisioning_cray.py

```diff
@@ -449,15 +500,536 @@ def test_last_used_time(self):
+        self.server.log_match(
+            "/opt/cray/capmc/default/bin/capmc node_off", starttime=t)
+        self.server.expect(NODE, {'state': 'sleep'}, id=self.names[0])
+        a = {"node_idle_limit": "1800", 'min_node_down_delay': '30'}
+        self.modify_hook_config(attrs=a, id='PBS_power')
+
+        jid = self.submit_job(1000, {'Resource_List.vnode': self.names[0],
+                                     'Resource_List.vntype': 'cray_compute'})
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+        t = int(time.time())
+        self.logger.info("Waiting for 1 min to poweron 1st node")
+        time.sleep(60)
+        self.server.log_match(
+            "/opt/cray/capmc/default/bin/capmc node_on", starttime=t)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+        self.server.expect(NODE, {'state': 'job-exclusive'}, id=self.names[0])
+
+    @timeout(900)
+    def test_power_on_ramp_rate_nodes(self):
+        """
+        Test when a both ramp rate and power on off is enabled,
+        power_on_off_enable will override and nodes will be powered off
+        and powered on.
+        """
+        self.scheduler.set_sched_config({'strict_ordering': 'True ALL'})
+        self.server.manager(MGR_CMD_SET, NODE, {
+                            "poweroff_eligible": True}, self.names[0])
+        a = {"power_on_off_enable": True,
+             "power_ramp_rate_enable": True, 'node_idle_limit': '30'}
+        self.modify_hook_config(attrs=a, id='PBS_power')
+        a = {'freq': 30}
+        self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
+                            sudo=True)
+        a = {'enabled': 'True'}
+        self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
+                            sudo=True)
+        t = int(time.time())
+        self.logger.info("Waiting for 1 min to poweroff 1st node")
+        time.sleep(60)
+        self.server.log_match(
+            "power_on_off_enable is over-riding power_ramp_rate_enable",
+            starttime=t)
+        self.server.log_match(
+            "/opt/cray/capmc/default/bin/capmc node_off", starttime=t)
+        self.server.expect(NODE, {'state': 'sleep'}, id=self.names[0])
+        a = {"node_idle_limit": "1800", 'min_node_down_delay': '30'}
+        self.modify_hook_config(attrs=a, id='PBS_power')
+
+        jid = self.submit_job(1000, {'Resource_List.vnode': self.names[0],
+                                     'Resource_List.vntype': 'cray_compute'})
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+        t = int(time.time())
+        self.logger.info("Waiting for 1 min to poweron 1st node")
+        time.sleep(60)
+        self.server.log_match(
+            "/opt/cray/capmc/default/bin/capmc node_on", starttime=t)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+        self.server.expect(NODE, {'state': 'job-exclusive'}, id=self.names[0])
+
+    @timeout(1500)
```

Comment:
I think 1500 is a bit excessive for this test.  You only sleep 1.5m during the test.